pycbc/inference/sampler/snowline.py

```diff
@@ -0,0 +1,166 @@
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides classes and functions for using the snowline sampler
+packages for parameter estimation.
+"""
+
+import sys
+import logging
+
+from pycbc.inference.io.snowline import SnowlineFile
+from pycbc.io.hdf import dump_state
+from pycbc.pool import use_mpi
+from .base import (BaseSampler, setup_output)
+from .base_cube import setup_calls
+
+
+#
+# =============================================================================
+#
+#                                   Samplers
+#
+# =============================================================================
+#
+
+class SnowlineSampler(BaseSampler):
+    """This class is used to construct an Snowline sampler from the snowline
+    package.
+
+    Parameters
+    ----------
+    model : model
+        A model from ``pycbc.inference.models``
+    """
+    name = "snowline"
+    _io = SnowlineFile
+
+    def __init__(self, model, **kwargs):
+        super().__init__(model)
+
+        import snowline
+        log_likelihood_call, prior_call = setup_calls(model, copy_prior=True)
+
+        self._sampler = snowline.ReactiveImportanceSampler(
+            list(self.model.variable_params),
+            log_likelihood_call,
+            transform=prior_call)
+
+        do_mpi, _, rank = use_mpi()
+        self.main = (not do_mpi) or (rank == 0)
+        self.nlive = 0
+        self.ndim = len(self.model.variable_params)
+        self.kwargs = kwargs
+
+    def run(self):
+        self.result = self._sampler.run(**self.kwargs)
+        if not self.main:
+            sys.exit(0)
+        self._sampler.print_results()
+
+    @property
+    def io(self):
+        return self._io
+
+    @property
+    def niterations(self):
+        return self.result['niter']
+
+    @classmethod
+    def from_config(cls, cp, model, output_file=None, **kwds):
+        """
+        Loads the sampler from the given config file.
+        """
+        skeys = {}
+        opts = {'num_global_samples': int,
+                'num_gauss_samples': int,
+                'max_ncalls': int,
+                'min_ess': int,
+                'max_improvement_loops': int
+                }
+        for opt_name in opts:
+            if cp.has_option('sampler', opt_name):
+                value = cp.get('sampler', opt_name)
+                skeys[opt_name] = opts[opt_name](value)
+        inst = cls(model, **skeys)
+
+        do_mpi, _, rank = use_mpi()
+        if not do_mpi or (rank == 0):
+            setup_output(inst, output_file)
+        return inst
+
+    def checkpoint(self):
+        pass
+
+    def resume_from_checkpoint(self):
+        pass
```

Comment:
Noted. It's not implemented at the moment. I'm not sure right now how to hook into the underlying code, but I would guess it might be possible in a similar way to dynesty (e.g. pickle) at the very least.